### PR TITLE
performance test workflow improvements

### PR DESCRIPTION
**Pull request by jeffreyscarpenter**
Scheduled runs of the performance test workflow do not run correctly due to malformed NoSQLBench command - `inputs` variable is only available for manually triggered workflows.

Also adding the `http_jsonapi_search_filter_sort` test to the list of tests to execute, and other minor improvements to workflows.

### CODE REVIEW

File: .github/workflows/performance-testing.yaml
Comment (ivansenic) [inline .github/workflows/performance-testing.yaml:63]: should we target different keyspace?

**Review by ivansenic — APPROVED** [review]
Looks good, but we need a green action run on this branch before we are merging this..

File: nosqlbench/http-jsonapi-crud-basic.yaml
Comment (ivansenic) [inline nosqlbench/http-jsonapi-crud-basic.yaml:3]: * we should also probably update this and remove the `auth_token` now as it seems invalid
* is driver needed at all?
* furthermore, why not having localhost here as default hostname
* and maybe also have `docscount=` and `connections=` in parameters

Just in order I can copy this line and run locally..

File: nosqlbench/http-jsonapi-search-basic.yaml
Comment (ivansenic) [inline nosqlbench/http-jsonapi-search-basic.yaml:1]: Does not run with `nb5 5.17.3-release`:

```
$ nb5 -v --report-csv-to logs --log-histograms logs/hdr_metrics.log http-jsonapi-search-basic jsonapi_host=localhost docscount=10000 connections=25 rate=250
     742 INFO  [main] NBCLI        Configured scenario log at logs/scenario_20230502_132422_664.log
     750 INFO  [main] NBCLI        Running NoSQLBench Version 5.17.3-release
     751 INFO  [main] NBCLI        command-line: -v --report-csv-to logs --log-histograms logs/hdr_metrics.log http-jsonapi-search-basic jsonapi_host=localhost docscount=10000 connections=25 rate=250
    2657 INFO  [main] NBCLI        client-hardware: {"processor-name":"Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz","heap-max-GiB":"7.78","if-speeds":["0Mib","10Gib"],"physical-cores":"6","max-frequency-ghz":"0.01","logical-cores":"12","sockets":"1","memory-GiB":"31.10"}
    2848 INFO  [main] OpsLoader    workload URI: 'file:///home/ise/workspace/datastax/stargate/jsonapi/nosqlbench/http-jsonapi-search-basic.yaml'
    2911 ERROR [main] ERRORHANDLER internal error: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
	at io.nosqlbench.engine.api.scenarios.NBCLIScenarioParser.parseScenarioCommand(NBCLIScenarioParser.java:157)
	at io.nosqlbench.engine.cli.NBCLICommandParser.parse(NBCLICommandParser.java:87)
	at io.nosqlbench.engine.cli.NBCLIOptions.parseAllOptions(NBCLIOptions.java:631)
	at io.nosqlbench.engine.cli.NBCLIOptions.<init>(NBCLIOptions.java:245)
	at io.nosqlbench.engine.cli.NBCLIOptions.<init>(NBCLIOptions.java:236)
	at io.nosqlbench.engine.cli.NBCLI.applyDirect(NBCLI.java:248)
	at io.nosqlbench.engine.cli.NBCLI.apply(NBCLI.java:114)
	at io.nosqlbench.engine.cli.NBCLI.main(NBCLI.java:94)
Scenario stopped due to error. See logs for details.

```

File: nosqlbench/http-jsonapi-search-filter-sort.yaml
Comment (ivansenic) [inline nosqlbench/http-jsonapi-search-filter-sort.yaml:1]: same here:

```
$ nb5 -v --report-csv-to logs --log-histograms logs/hdr_metrics.log http-jsonapi-search-filter-sort jsonapi_host=localhost docscount=10000 connections=25 rate=250
     760 INFO  [main] NBCLI        Configured scenario log at logs/scenario_20230502_132401_299.log
     767 INFO  [main] NBCLI        Running NoSQLBench Version 5.17.3-release
     768 INFO  [main] NBCLI        command-line: -v --report-csv-to logs --log-histograms logs/hdr_metrics.log http-jsonapi-search-filter-sort jsonapi_host=localhost docscount=10000 connections=25 rate=250
    2653 INFO  [main] NBCLI        client-hardware: {"memory-GiB":"31.10","processor-name":"Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz","heap-max-GiB":"7.78","if-speeds":["0Mib","10Gib"],"physical-cores":"6","max-frequency-ghz":"0.01","logical-cores":"12","sockets":"1"}
    2845 INFO  [main] OpsLoader    workload URI: 'file:///home/ise/workspace/datastax/stargate/jsonapi/nosqlbench/http-jsonapi-search-filter-sort.yaml'
    2910 ERROR [main] ERRORHANDLER internal error: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
	at io.nosqlbench.engine.api.scenarios.NBCLIScenarioParser.parseScenarioCommand(NBCLIScenarioParser.java:157)
	at io.nosqlbench.engine.cli.NBCLICommandParser.parse(NBCLICommandParser.java:87)
	at io.nosqlbench.engine.cli.NBCLIOptions.parseAllOptions(NBCLIOptions.java:631)
	at io.nosqlbench.engine.cli.NBCLIOptions.<init>(NBCLIOptions.java:245)
	at io.nosqlbench.engine.cli.NBCLIOptions.<init>(NBCLIOptions.java:236)
	at io.nosqlbench.engine.cli.NBCLI.applyDirect(NBCLI.java:248)
	at io.nosqlbench.engine.cli.NBCLI.apply(NBCLI.java:114)
	at io.nosqlbench.engine.cli.NBCLI.main(NBCLI.java:94)
Scenario stopped due to error. See logs for details.

```

File: nosqlbench/http-jsonapi-crud-basic.yaml
Comment (ivansenic) [inline nosqlbench/http-jsonapi-crud-basic.yaml:1]: trying to run this locally, produces strange results.. the scenario ends correctly and no errors are reported, but on the server-side metrics I am seeing a lot of client errors:

```
# HELP http_server_requests_seconds  
# TYPE http_server_requests_seconds summary
http_server_requests_seconds_count{method="POST",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/v1/{namespace}/{collection}",} 40000.0
http_server_requests_seconds_sum{method="POST",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/v1/{namespace}/{collection}",} 620.394043765
http_server_requests_seconds_count{method="POST",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/v1",} 2.0
http_server_requests_seconds_sum{method="POST",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/v1",} 1.928188204
http_server_requests_seconds_count{method="POST",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/v1/{namespace}",} 4.0
http_server_requests_seconds_sum{method="POST",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/v1/{namespace}",} 14.483613413
http_server_requests_seconds_count{method="POST",module="sgv2-jsonapi",outcome="CLIENT_ERROR",status="RESET",uri="/v1/{namespace}/{collection}",} 40000.0
http_server_requests_seconds_sum{method="POST",module="sgv2-jsonapi",outcome="CLIENT_ERROR",status="RESET",uri="/v1/{namespace}/{collection}",} 617.666413847
http_server_requests_seconds_count{method="POST",module="sgv2-jsonapi",outcome="CLIENT_ERROR",status="RESET",uri="/v1/{namespace}",} 4.0
http_server_requests_seconds_sum{method="POST",module="sgv2-jsonapi",outcome="CLIENT_ERROR",status="RESET",uri="/v1/{namespace}",} 12.24249949
http_server_requests_seconds_count{method="GET",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/favicon.ico",} 1.0
http_server_requests_seconds_sum{method="GET",module="sgv2-jsonapi",outcome="SUCCESS",status="200",uri="/favicon.ico",} 0.012527204
http_server_requests_seconds_count{method="POST",module="sgv2-jsonapi",outcome="CLIENT_ERROR",status="RESET",uri="/v1",} 2.0
http_server_requests_seconds_sum{method="POST",module="sgv2-jsonapi",outcome="CLIENT_ERROR",status="RESET",uri="/v1",} 1.926456839
```

Comment (ivansenic) [reply nosqlbench/http-jsonapi-crud-basic.yaml:1]: Created this: https://github.com/stargate/jsonapi/issues/411

**Review by ivansenic — CHANGES_REQUESTED** [review]
I tried to run some of these locally in preparation for the smoke and soak tests I need to implemented.. Seems that there are some issues which we should try to solve, but maybe in a separate PR/issue.. Anyway will try to document it here:

* I saw those client errors on the server side.. I am not sure what's happening here, considering we have a body assertion.. Could be something related to the `nb5`, needs investigation
* deletes are doing random deletes, which actually leaves data in the table after run, which could mean that our "how much data check at the end of the test" could just be reflecting the about of the leftover documents in the table and not the actual size for all documents
  ```
  {
    "status": {
      "count": 3704
    }
  }
  ```
  
Let's for now just fix here the small comments and that issue in running those two new tests..

File: nosqlbench/http-jsonapi-search-basic.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-search-basic.yaml:1]: This scenario tests functionality like `gt` and `lt` that is not implemented yet. I will add a comment to clarify this is not a working scenario yet.

Comment (ivansenic) [reply nosqlbench/http-jsonapi-search-basic.yaml:1]: have a look at the stack trace, it has nothing to do with tests, seems YAML is wrongs:

```
 2911 ERROR [main] ERRORHANDLER internal error: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class java.lang.String (java.util.LinkedHashMap and java.lang.String are in module java.base of loader 'bootstrap')
```

File: nosqlbench/http-jsonapi-crud-basic.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-crud-basic.yaml:1]: Thanks, we can continue investigation there.

File: nosqlbench/http-jsonapi-search-basic.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-search-basic.yaml:1]: I have updated the scenario to contain a new TODO note, and updated the scenario definitions to correctly reference operations they are intended to run. If you run the default scenario now it will run to completion, correctly indicating the calls that had error messages due to syntax that is not yet implemented.

File: nosqlbench/http-jsonapi-search-filter-sort.yaml
Comment (jeffreyscarpenter) [reply nosqlbench/http-jsonapi-search-filter-sort.yaml:1]: Fixed. The workload will now run to completion but will indicate errors when using behavior not yet implemented.

**Comment by jeffreyscarpenter** [comment]
@ivansenic please re-review. Ok to merge if it looks good.

**Review by ivansenic — APPROVED** [review]

File: .github/workflows/performance-testing.yaml
Comment (ivansenic) [reply .github/workflows/performance-testing.yaml]: Seems still not fixed?

Comment (jeffreyscarpenter) [reply .github/workflows/performance-testing.yaml]: looks fixed to me.

**Pull request merged**
